lib/posix/pthread.c

```diff
@@ -685,13 +682,20 @@ int pthread_join(pthread_t pthread, void **status)
 		t->detachstate = PTHREAD_CREATE_DETACHED;
 	}
 
-	if (ret != 0) {
+	switch (ret) {
+	case ESRCH:
+		LOG_ERR("Pthread %p has already been joined", &t->thread);
 		return ret;
+	case EINVAL:
+		LOG_ERR("Pthread %p is not a joinable", &t->thread);
+		return ret;
+	case 0:
+		break;
```

Comment:
Just wondering whether it wouldn't be as good to just do:
```
LOG_ERR("Pthraead %p %s", &t->thread, msg)
```
and set msg according to ret. Probably would save us some flash.

Reply:
yep, that would work. would probably be fewer lines of code too